include/seqan3/search/dream_index/interleaved_bloom_filter.hpp

```diff
@@ -525,11 +451,157 @@ class interleaved_bloom_filter
         archive(bin_words);
         archive(hash_funs);
         archive(data);
-        result_buffer.resize(bins);
     }
     //!\endcond
 };
 
+/*!\brief Manages queries for the seqan3::interleaved_bloom_filter.
+ * \attention Calling seqan3::interleaved_bloom_filter::increase_bin_number_to on `ibf` invalidates the bulk_agent.
+ *
+ * \details
+ *
+ * ### Example
+ *
+ * \include test/snippet/search/dream_index/bulk_agent_constructor.cpp
+ */
+template <data_layout data_layout_mode>
+class interleaved_bloom_filter<data_layout_mode>::bulk_agent
+{
+private:
+    //!\brief The type of the augmented seqan3::interleaved_bloom_filter.
+    using ibf_t = interleaved_bloom_filter<data_layout_mode>;
+
+    //!\brief A pointer to the augmented seqan3::interleaved_bloom_filter.
+    ibf_t const * ibf_ptr;
```

Comment:
```suggestion
    ibf_t const * ibf_ptr{nullptr};
```
or is this equivalent?

Reply:
It's equivalent but I like the explicit version more